BepInEx/Bootstrap/Chainloader.cs

```diff
@@ -48,7 +48,21 @@ private static bool IsHeadless
 				var prop = AccessTools.PropertyGetter(typeof(Application), "isBatchMode");
 				if (prop != null)
 					return (bool) prop.Invoke(null, null);
-				return SystemInfo.graphicsDeviceID == 0;
+				try {
+					return SystemInfo.graphicsDeviceID == 0;
+				}
+				catch (MissingMethodException)
+				{
+					return false; // Unity 2022.3.62f3 for example doesn't have such method
+				}
+				catch (MissingFieldException)
+				{
+					return false; // Unity 2022.3.62f3 for example doesn't have such field
+				}
+				catch (MissingMemberException)
+				{
+					return false; // Unity 2022.3.62f3 for example doesn't have such member
+				}
```

Comment:
What exactly is the existing code throwing? If the method/property is missing then I would expect this trycatch to not prevent the crash, since it would crash when parsing the method body rather than on execution. I suggest switching to using AccessTools.PropertyGetter in that case. This will not work if the property getter exists but throws when called!

```suggestion
				// Some Unity builds may miss this property, see 
				// https://docs.unity3d.com/2022.3/Documentation/ScriptReference/SystemInfo-graphicsDeviceID.html
				var prop2 = AccessTools.PropertyGetter(typeof(SystemInfo), "graphicsDeviceID");
				if (prop2 != null)
					return ((int) prop2.Invoke(null, null)) == 0;
				
				return false;
```

Reply:
> What exactly is the existing code throwing? If the method/property is missing then I would expect this trycatch to not prevent the crash, since it would crash when parsing the method body rather than on execution. I suggest switching to using AccessTools.PropertyGetter in that case. This will not work if the property getter exists but throws when called!

middingmethod